lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -44,35 +46,34 @@
   void onRestart();
 
   /**
-   * Indicate that the Driver restart is complete. It is meant to be called exactly once during a restart and never
-   * during the ininital launch of a Driver.
+   * @return whether restart is completed.
    */
-  void setRestartCompleted();
+  boolean isRestartCompleted();
 
   /**
-   * @return the number of Evaluators expected to check in from a previous run.
+   * @return the Evaluators expected to check in from a previous run.
    */
-  int getNumPreviousContainers();
-
+  Set<String> getPreviousEvaluators();
```

Comment:
Please rename to `getPreviousEvaluatorIds` to make the return type more obvious.